### PR TITLE
Firebase

### PR DESCRIPTION
**Pull request by reemsh2000**
starting with firebase signup

### CODE REVIEW

**Review by mazenchami — COMMENTED** [review]
@reemsh2000 as far as logic goes, it seems like you have it done correctly. Take a look at this [example](https://docs.expo.dev/versions/latest/sdk/firebase-recaptcha/#example-usage). This works with phone, try to emulate it in your code and see if that works.

**Comment by reemsh2000** [comment]
> @reemsh2000 as far as logic goes, it seems like you have it done correctly. Take a look at this [example](https://docs.expo.dev/versions/latest/sdk/firebase-recaptcha/#example-usage). This works with phone, try to emulate it in your code and see if that works.

It worked, thank you very much Mazen ❤️❤️

**Review by Nizar7zak — APPROVED** [review]
Great Work Reem

File: app/hooks/useFirebase.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import { useState, useRef } from "react";
+import { PhoneAuthProvider, signInWithCredential } from "firebase/auth";
+import { auth } from "../../firebase/firebase";
+
+const useFirebase = () => {
+  const [phoneNumber, setPhoneNumber] = useState();
+  const [verificationCode, setVerificationCode] = useState();
+  const [verificationId, setVerificationId] = useState();
+  const [message, showMessage] = useState();
```
Comment (mazenchami) [inline app/hooks/useFirebase.js:9]: [nit pick] I personally feel the `callback` in `useState` should start with `set...` so this line would be (if you agree with this suggestion, you'll need to update lines 19, 21, 33, & 37): 
```suggestion
  const [message, setShowMessage] = useState();
```

File: app/screens/Login.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import useFirebase from "../hooks/useFirebase";
+import { app } from "../../firebase/firebase";
+export default function LoginScreen() {
+
+  const { options } = app;
+  const attemptInvisibleVerification = false;
+  const authByPhone = useFirebase();
+
+  return (
+    <View style={{ padding: 20, marginTop: 50 }}>
```
Comment (mazenchami) [inline app/screens/Login.js:23]: [nit picks] Another personal recommendation, move in-line styles to `StyleSheet.create`

**Review by mazenchami — APPROVED** [review]
I know you already merged this branch, but here are a couple comments for ideas. Feel free to assign me as a reviewer on other PRs and I can give you more insight like this.

File: app/screens/Login.js (hunk 2 above)
Comment (reemsh2000) [reply app/screens/Login.js:23]: can you tell us where is the right place to put firebase configuration, please?

Comment (mazenchami) [reply app/screens/Login.js:23]: for the styles, I would keep them in the same file.

Comment (mazenchami) [reply app/screens/Login.js:23]: take a look at the first example [here](https://reactnative.dev/docs/0.66/style)

Comment (Nizar7zak) [reply app/screens/Login.js:23]: Hi Mazen, Thank you  for your support, even if I want configuration in another files?

Comment (mazenchami) [reply app/screens/Login.js:23]: @Nizar7zak not sure what you mean. Are you talking about the `userFirebase` hook or styles?

I was just mentioning to move this files styles to the `StyleSheet.create` function in the same file. I usually tend to have styles per file and not reusable styles as that can get overly complicated.

If easier, here is my Discord: mazenchami#9806. We can chat there or even voice call

Comment (Nizar7zak) [reply app/screens/Login.js:23]: Yes I asked about useFireBase hook?

Comment (mazenchami) [reply app/screens/Login.js:23]: @Nizar7zak regarding `useFirebase` hook, I think you have everything in the right way. I would maybe the folder `/firebase` to `/app/firebase` and keep everything in your `app` folder

**Pull request merged**